File: annotation-processor-testing/src/main/java/com/tickaroo/tikxml/annotationprocessing/elementlist/EmptyTagList.java
package com.tickaroo.tikxml.annotationprocessing.elementlist;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import com.tickaroo.tikxml.annotationprocessing.element.EmptyTag;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class EmptyTagList {

  @Element
  List<EmptyTag> tags;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof EmptyTagList)) return false;

    EmptyTagList that = (EmptyTagList) o;

    return tags != null ? tags.equals(that.tags) : that.tags == null;
  }

  @Override public int hashCode() {
    return tags != null ? tags.hashCode() : 0;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/elementlist/EmptyTagList.kt
package com.tickaroo.tikxml.annotationprocessing.elementlist

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml
import com.tickaroo.tikxml.annotationprocessing.element.EmptyTag

/**
 * @author Hannes Dorfmann
 */
@Xml
class EmptyTagList {

    @Element
    var tags: List<EmptyTag>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is EmptyTagList) return false

        val that = other as EmptyTagList?

        return if (tags != null) tags == that!!.tags else that!!.tags == null
    }

    override fun hashCode(): Int {
        return if (tags != null) tags!!.hashCode() else 0
    }
}
